pkg/controller/workload/workload_processor.go

```diff
@@ -89,6 +93,28 @@ func newAckRequest(rsp *service_discovery_v3.DeltaDiscoveryResponse) *service_di
 	}
 }
 
+func GetIdentityByUid(workloadUid string) string {
+	workload := cache.WorkloadCache.GetWorkloadByUid(workloadUid)
+	if workload == nil {
+		log.Errorf("workload %v not found", workloadUid)
+		return ""
+	}
+
+	return spiffe.Identity{
+		TrustDomain:    workload.TrustDomain,
+		Namespace:      workload.Namespace,
+		ServiceAccount: workload.ServiceAccount,
+	}.String()
+}
+
+func isManagedWorkload(workloadUid string) bool {
```

Comment:
We can get node name from env `NODE_NAME`,  and then compare workload.node wirth it.